### PR TITLE
[JBPM-9353] Kafka WIH manages exceptions

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**: [JBPM-9353](https://issues.redhat.com/browse/JBPM-9353)

### CODE REVIEW

**Comment by gmunozfe** [comment]
@elguardian please review.
Finally, I put a 5 seconds deadline in the future.get to avoid getting stuck there

File: kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java [hunk 1]
```diff
@@ -109,22 +112,10 @@ public void executeWorkItem(WorkItem workItem,
...
-                LOG.error("Kafka error",
-                          e);
-                producer.flush();
-                results.put(RESULTS_VALUE,
-                            "failure");
-            }
+            Future<RecordMetadata> future = producer.send(new ProducerRecord(topic, key, value));
+            RecordMetadata metadata = future.get(5, TimeUnit.SECONDS);
+            
+            manager.completeWorkItem(workItem.getId(), null);
```
Comment (elguardian) [inline kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java:118]: broke backward compatbility. requires the RESULTS_VALUE.... as completed.

Comment (elguardian) [inline kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java:118]:
```suggestion
            manager.completeWorkItem(workItem.getId(), Collections.singletonMap(RESULTS_VALUE, "success"));
```

Comment (gmunozfe) [reply kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java]: OK, put it back again

Comment (gmunozfe) [reply kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java]: Added again, and also covered in a unit test to check the right value
Thanks for review @elguardian !

**Comment by elguardian** [comment]
Jenkins retest this please.

**Pull request merged**